osu.Game/Beatmaps/LocalCachedBeatmapMetadataSource.cs

```diff
@@ -114,6 +114,26 @@ public bool TryLookup(BeatmapInfo beatmapInfo, [NotNullWhen(true)] out OnlineBea
                     }
                 }
             }
+            catch (SqliteException sqliteException)
+            {
+                // There have been cases where the user's local database is corrupt.
+                // Let's attempt to identify these cases and re-initialise the local cache.
+                switch (sqliteException.SqliteErrorCode)
+                {
+                    case 26: // SQLITE_NOTADB
+                    case 11: // SQLITE_CORRUPT
+                        // only attempt purge & re-download if there is no other refetch in progress
+                        if (cacheDownloadRequest != null)
+                            throw;
```

Comment:
This looks like a behavioural change? I think this should `return false`? If the intention was to rethrow to the `catch (Exception ex)` block below then I'm not sure that's how this works.

In an isolated console project, when I do something like

```csharp
try
{
   throw new InvalidOperationException();
}
catch (InvalidOperationException)
{
   throw;
}
catch
{
   Console.WriteLine("caught");
}
```

then the resulting program is not printing to the console, it's throwing the error:

```
Unhandled exception. System.InvalidOperationException: Operation is not valid due to the current state of the object.
   at Program.<Main>$(String[] args) in /home/dachb/Documents/Projekty/ConsoleApp1/ConsoleApp1/Program.cs:line 5
```

Reply:
right, that's unfortunate. i thought that's how it worked but i guess not.

Reply:
I made one more attempt.

Reply:
Seems better now yeah.